static/app/components/charts/barChart.tsx

```diff
@@ -12,25 +12,46 @@ export interface BarChartSeries
     Series {}
 
 export interface BarChartProps extends BaseChartProps {
+  barOpacity: number;
   series: BarChartSeries[];
   animation?: boolean;
+  hideZeros?: boolean;
   stacked?: boolean;
 }
 
 export function transformToBarSeries({
+  barOpacity,
   series,
   stacked,
+  hideZeros = false,
   animation,
-}: Pick<BarChartProps, 'series' | 'stacked' | 'animation'>) {
+}: Pick<BarChartProps, 'barOpacity' | 'hideZeros' | 'series' | 'stacked' | 'animation'>) {
   return series.map(({seriesName, data, ...options}) =>
     BarSeries({
       name: seriesName,
       stack: stacked ? 'stack1' : undefined,
       data: data?.map(({value, name, itemStyle}) => {
         if (itemStyle === undefined) {
-          return [name, value];
+          return {
+            value: [name, value],
+            itemStyle: value === 0 && hideZeros ? {opacity: 0} : {opacity: barOpacity},
+            emphasis:
+              value === 0 && hideZeros
+                ? {itemStyle: {opacity: 0}}
+                : {itemStyle: {opacity: 1}},
+          };
         }
-        return {value: [name, value], itemStyle};
+        return {
+          value: [name, value],
+          itemStyle:
+            value === 0 && hideZeros
+              ? {...itemStyle, opacity: 0} // Don't show bars when value = 0 (when hideZeros is enabled)
+              : {...itemStyle, opacity: barOpacity},
+          emphasis:
+            value === 0 && hideZeros
+              ? {itemStyle: {opacity: 0}}
+              : {itemStyle: {opacity: 1}},
+        };
```

Comment:
New logic that makes the bar invisible when the value of the bar is 0. 

The existing events graph in the issue stream shows a bar with minimum height even if the value is 0. This is because the existing graph does not show the x axis, so if we hid the bar with 0 values, the graph would look very disjointed.